packages/kbn-custom-integrations/src/state_machines/services/integrations_client.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { HttpSetup } from '@kbn/core/public';
+import { EPM_API_ROUTES } from '@kbn/fleet-plugin/common';
+import * as rt from 'io-ts';
+import { i18n } from '@kbn/i18n';
+import { decodeOrThrow } from '@kbn/io-ts-utils';
+import {
+  AuthorizationError,
+  customIntegrationOptionsRT,
+  DecodeError,
+  integrationNameRT,
+  IntegrationNotInstalledError,
+  NamingCollisionError,
+  UnknownError,
+} from '../../types';
+
+const GENERIC_CREATE_ERROR_MESSAGE = i18n.translate(
+  'customIntegrationsPackage.genericCreateError',
+  {
+    defaultMessage: 'Unable to create an integration',
+  }
+);
+
+const GENERIC_DELETE_ERROR_MESSAGE = i18n.translate(
+  'customIntegrationsPackage.genericDeleteError',
+  {
+    defaultMessage: 'Unable to delete integration',
+  }
+);
+
+/**
+ * Constants
+ */
+const CUSTOM_INTEGRATIONS_URL = EPM_API_ROUTES.CUSTOM_INTEGRATIONS_PATTERN;
+const DELETE_PACKAGE_URL = EPM_API_ROUTES.DELETE_PATTERN;
+
+export interface IIntegrationsClient {
+  createCustomIntegration(
+    params?: CreateCustomIntegrationRequestQuery
+  ): Promise<CreateCustomIntegrationValue>;
+  deleteCustomIntegration(
+    params?: DeleteCustomIntegrationRequestQuery
+  ): Promise<DeleteCustomIntegrationResponse>;
+}
+
+export class IntegrationsClient implements IIntegrationsClient {
+  constructor(private readonly http: HttpSetup) {}
+
+  public async createCustomIntegration(
+    params: CreateCustomIntegrationRequestQuery
+  ): Promise<CreateCustomIntegrationValue> {
+    try {
+      const response = await this.http.post(CUSTOM_INTEGRATIONS_URL, {
+        version: '2023-10-31',
+        body: JSON.stringify(params),
+      });
+
+      const data = decodeOrThrow(
+        createCustomIntegrationResponseRT,
+        (message: string) =>
+          new DecodeError(`Failed to decode create custom integration response: ${message}"`)
+      )(response);
+
+      return Promise.resolve({
```

Comment:
**nit**: wrapping the returned value with Promise.resolve shouldn't be necessary as the function is already `async` and will return a promise automatically